sklearn/preprocessing/_discretization.py

```diff
@@ -374,15 +374,7 @@ def transform(self, X):
 
         bin_edges = self.bin_edges_
         for jj in range(Xt.shape[1]):
-            # Values which are close to a bin edge are susceptible to numeric
-            # instability. Add eps to X so these values are binned correctly
-            # with respect to their decimal truncation. See documentation of
-            # numpy.isclose for an explanation of ``rtol`` and ``atol``.
-            rtol = 1.0e-5
-            atol = 1.0e-8
-            eps = atol + rtol * np.abs(Xt[:, jj])
-            Xt[:, jj] = np.digitize(Xt[:, jj] + eps, bin_edges[jj][1:])
-        np.clip(Xt, 0, self.n_bins_ - 1, out=Xt)
+            Xt[:, jj] = np.searchsorted(bin_edges[jj][1:-1], Xt[:, jj], side="right")
```

Comment:
Since we are changing behavior slightly anyways, I would prefer to use the default `side="left"` if it passes our tests.

Reply:
Changing to "left" fails 21 tests

Reply:
Okay let's keep it as right